hashes/src/sha256.rs

```diff
@@ -125,11 +125,6 @@ impl Midstate {
     /// Length of the midstate, in bytes.
     const LEN: usize = 32;
 
-    /// Flag indicating whether user-visible serializations of this hash
-    /// should be backward. For some reason Satoshi decided this should be
-    /// true for `Sha256dHash`, so here we are.
```

Comment:
Funny, this says sha256d but it's in non-d module. I'd say it was broken already.

What kinda annoys me that Satoshi did some weird thing and we've added it to midstate as well. Is this standardized across Bitcoin ecosystem?

Reply:
If the midstate is serialized backward then probably it's because we do this in Elements. I'm not sure where else (if anywhere) midstates are serialized in bitcoin. I'm a little hesitant to change it now, though now that we've moved to "you should be using newtypes for all hashes that mean something" I don't feel very strongly.

Reply:
Ok, I honestly thought serializing midstate is so obscure nobody is doing it in practice. Let's keep it.

Reply:
Well, properly speaking the "midstate" is actually an asset ID, and we have a newtype for it. But the newtype `Display` impl just passes through to the underlying midstate type.

We have unit tests for this, so I think that if we're breaking the API anyway we could take the opportunity to also change the direction of serialization (and then rust-elements will need to do a bit of extra work). But I'd rather not.

I suppose, we could change the comment on this field to reflect that it's about Elements' asset IDs rather than directly about sha256d ... but it's all the same original sin of storing hashes using a LE `uint256` type.

Reply:
Do we even need to be able to `Display` a `Midstate`, can we just `Debug` and have to/from and getters and be done with it?

Reply:
Flagging this as unresolved however I hacked it up based on my personal understanding/ide that `Midstate` is super niche, we should just provide `into_parts` and `Debug` and let niche users do what they want with it.

Reply:
Nice, concept ACK dropping `Display` in favor of `into_parts` (and a pair of specific accessors, for convenience) and `debug`.